### PR TITLE
swagger.yaml changes

### PR DESCRIPTION
**Pull request by pujavs**
Updated oxAuth_swagger.yaml  based on review feedback

### CODE REVIEW

**Comment by pujavs** [comment]
Hi Yuriy, Whilst awaiting your review, I am continuing on same lines as this swagger.xml. Once complete will try to generate  client.

File: docs/oxAuth_swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,3382 @@
...
+                  example: false      
+                spontaneous_scopes:
+                  type: array
+                  description: List of spontaneous scopes 
+                  items:
+                    type: string
+                run_introspection_script_before_access_token_as_jwt_creation_and_include_claims:
+                  type: boolean
+                  description: Boolean value with default value false. If true and access_token_as_jwt=true then run introspection script 
+                              and
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: describe permission object https://docs.kantarainitiative.org/uma/wg/rec-oauth-uma-federated-authz-2.0.html#uma-bearer-token-profile

Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: token_type_hint is not required

Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: describe permission object as explained above

Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: Lets remove this endpoint.

File: docs/oxAuth_swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,3382 @@
...
+      responses:
+        200:
+          description: OK
+          content: {}
+        400:
+          $ref: '#/components/responses/InvalidRequest'        
+        401:
+          $ref: '#/components/responses/Unauthorized'
+        500:
+          $ref: '#/components/responses/InternalServerError'
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: **Added the same now**
I had shared my finding on this in https://github.com/GluuFederation/oxAuth/pull/1276
I could see @ApiResponses- 302 specified in interface AuthorizeRestWebService but when I went through AuthorizeRestWebServiceImpl.java code and found various AuthorizeErrorResponseType , but couldn't find its mapping to 302 (RedirectUriResponse-> ErrorResponseFactory-> ErrorMessages- >RedirectUri) 
and  hence had not mention it thinking that the its not in sync with the actual mplementation.

File: docs/oxAuth_swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,3382 @@
...
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: **Added the same now**
I had shared my finding on this in https://github.com/GluuFederation/oxAuth/pull/1276
I could see @ApiResponses- 302 specified in interface AuthorizeRestWebService but when I went through AuthorizeRestWebServiceImpl.java code and found various AuthorizeErrorResponseType , but couldn't find its mapping to 302 (RedirectUriResponse-> ErrorResponseFactory-> ErrorMessages- >RedirectUri) 
and  hence had not mention it thinking that the its not in sync with the actual mplementation.

File: docs/oxAuth_swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,3382 @@
...
+      operationId: post-introspection      
+      requestBody:
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              required:
+              - token
+              - token_type_hint
+              - response_as_jwt
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done

File: docs/oxAuth_swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,3382 @@
...
+          example: Bearer 23410913-abewfq.123483
+        - name: token
+          in: query
+          required: true    
+          schema:
+            type : string
+          example: token=2YotnFZFEjr1zCsicMWpAA
+        - name: token_type_hint
+          in: query
+          required: true
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Removed token_type_hint

File: docs/oxAuth_swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,3382 @@
...
+          description: ID Token previously issued by the Authorization Server being passed as a hint about the End-User.            
+          schema:
+            type : string
+        - name: response_as_jwt
+          in: query
+          required: false
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Removed 'required' attribute for response_as_jwt

File: docs/oxAuth_swagger.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Did you mean the grammar/tense  in the decription to be modified as i do not see any other issue or incomplete sentence.
**Old description**:  Boolean value with default value false. If true and access_token_as_jwt=true then run introspection script  and transfer claims into JWT.
**New description**: If true and access_token_as_jwt value is true then introspection script to be executed to transfer claims into JWT.  The default value is false.

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: The screenshot seems mismtach...
I have added description in /uma2-configuration endpoint for of permission_endpoint, resource_registration_endpoint and introspection_endpoint
 description: The endpoint URI at which the resource server requests permissions on the client's behalf.

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: The screenshot seems mismtach...
I have removed required attribute for token_type_hint from 
image

File: docs/oxAuth_swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: Please pay attention to how response builder is created.
See https://github.com/GluuFederation/oxAuth/blob/19eb222c5f841dee24ad08fdc54439526abbdd68/Server/src/main/java/org/gluu/oxauth/authorize/ws/rs/AuthorizeRestWebServiceImpl.java#L544

File: docs/oxAuth_swagger.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: Don't look to `@ApiResponses` annotations, look to implementation.

https://github.com/GluuFederation/oxAuth/blob/19eb222c5f841dee24ad08fdc54439526abbdd68/Server/src/main/java/org/gluu/oxauth/authorize/ws/rs/AuthorizeRestWebServiceImpl.java#L544

File: docs/oxAuth_swagger.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: I think it's comment in wrong line, you can ignore it.

Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: I gave you link to how permission object look in json. You have to define it in swagger spec, not just `type: object`. It doesn't say anything about internal structure of the object. It NOT about `description`, it's about missed definition of the permission object.

Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: When I reviewed it I think it wasn't mistmached. Maybe you updated PR and it shifted?

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: I have added description for permission object in response of   /rpt/status endpoint
image

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Which endpoint?

Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: see https://github.com/GluuFederation/oxAuth/pull/1285#discussion_r390934135

Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: image

[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: OK, it's in correct line, you have to open file to see correct line.

image

So it's about `_description`

Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: @pujavs please open reviewed file to see exact lines. https://github.com/GluuFederation/oxAuth/pull/1285/files

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done
image

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done
image

Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done.
Not sure why I am not seeing your review comments screenshot. All review screenshots baring the first 5 I see Line #1269 run_introspection_script_before_access_token_as_jwt_creation_and_include_claims:
image

**Comment by pujavs** [comment]
Yuriry, Do you think we are good on swagger spec for oxAuth.?
Request you to please guide me on further steps regrading the same

**Comment by pujavs** [comment]
Closing this pull request, as Yuiry has created related  PR#1289 to Arnab Dutta

**Comment by pujavs** [comment]
Closing this pull request, as Yuiry has created related PR#1289 to Arnab Dutta

**Comment by yuriyz** [comment]
@pujavs Actually Arnab got new assignments thus it would be nice if you can finish this task per comments in this PR.

File: docs/oxAuth_swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Got it.
Response 302 added

File: docs/oxAuth_swagger.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Response 302 added

File: docs/oxAuth_swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,3382 @@
...
+          description: OPTIONAL. Boolean value with default value false. If true, returns introspection response as JWT (signed based on client configuration used for authentication to Introspection Endpoint).
+          schema:
+            type : string
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified this for introspection end point line#578

File: docs/oxAuth_swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,3382 @@
...
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                title: IntrospectionResponse
+                description: meta-information about token
+                required:
+                - active
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Changes done to make 'active' as required.
Comparing the spec viz code IntrospectionWebService.introspect(..)  could not find two properties nbf and jti

File: docs/oxAuth_swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,3382 @@
...
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Changes done to make 'active' as required.
Comparing the spec viz code IntrospectionWebService.introspect(..) could not find two properties nbf and jti

File: docs/oxAuth_swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,3382 @@
...
+                required:
+                - keys
+                type: object
+                properties:
+                  keys:
+                    type: array
+                    items:
+                      type: string
+                    description: Each property in the key is defined by the JWK specification RFC 7517 Section 4 or, for algorithm-specific properties, in RFC 7518]
+                    example: '{"keys": [{"key1": "value1", "key2": "value2"}] }'
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Have done the modification.
Howver there were few difference that i am not sure like kty, use, alg, crv, family(AlgorithmFamily)  are string in spec where as objects in our code.
Please verify and confirm

File: docs/oxAuth_swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,3382 @@
...
+                jwks_uri:
+                  type: string
+                  description: URL for the Client's JSON Web Key Set (JWK) document containing key(s) that are used for signing requests
+                    to the OP. The JWK Set may also contain the Client's encryption
+                    keys(s) that are used by the OP to encrypt the responses to the
+                    Client. When both signing and encryption keys are made available,
+                    a use (Key Use) parameter value is required for all keys in the
+                    document to indicate each key's intended usage .
+                jwks:
+                  type: string
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: modified to type array referencing '#/components/schemas/JsonWebKey'

File: docs/oxAuth_swagger.yaml [hunk 12]
```diff
@@ -0,0 +1,3382 @@
...
+                    value. The semantics of the jwks parameter are the same as the
+                    jwks_uri parameter, other than that the JWK Set is passed by value,
+                    rather than by reference. This parameter is intended only to be
+                    used by Clients that, for some reason, are unable to use the jwks_uri
+                    parameter, for instance, by native applications that might not
+                    have a location to host the contents of the JWK Set. If a Client
+                    can use jwks_uri, it must not use jwks. One significant downside
+                    of jwks is that it does not enable key rotation. The jwks_uri
+                    and jwks parameters must not be used together.
+                  example: '{"key1": "value1", "key2": "value2"}'
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Modified to ->  example: '{ "keys" : [ { "e" : "AQAB", "n" : "gmlDX_mgMcHX.." ] }'

File: docs/oxAuth_swagger.yaml [hunk 13]
```diff
@@ -0,0 +1,3382 @@
...
+                jwks_uri:
+                  type: string
+                  description: URL for the Client's JSON Web Key Set (JWK) document containing key(s) that are used for signing requests
+                    to the OP. The JWK Set may also contain the Client's encryption
+                    keys(s) that are used by the OP to encrypt the responses to the
+                    Client. When both signing and encryption keys are made available,
+                    a use (Key Use) parameter value is required for all keys in the
+                    document to indicate each key's intended usage .
+                jwks:
+                  type: string
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Modified to type array referencing '#/components/schemas/JsonWebKey'

File: docs/oxAuth_swagger.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rearranged the details as requested

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Removed required: true for 'id_token_hint' param of /end_session endopint

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Modified to specify resource_scopes as required. Also modified the response object properties as per @JsonProperty(value...) in UmaResource class

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified -  I was getting error in swagger editor earlier with conflicting with swagger field name however strangely not facing the issue now

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified now, my bad i overlooked it earlier. Have now modified the object properties as per @JsonProperty(value...) in UmaResource class

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Have now modified the object properties as per @JsonProperty(value...) in UmaResource class

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified -  I was getting error in swagger editor earlier with conflicting with swagger field name however strangely not facing the issue now

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Modified to specify resource_scopes as required. Also modified the response object properties as per @JsonProperty(value...) in UmaResource class

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified -  I was getting error in swagger editor earlier with conflicting with swagger field name however strangely not facing the issue now

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Modified to specify resource_scopes as required. Also modified the remaining response object properties as per @JsonProperty(value...) in UmaResource class

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Rectified

[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: rectified _description to description

File: docs/oxAuth_swagger.yaml [hunk 14]
```diff
@@ -0,0 +1,3426 @@
...
+                  scopes:
+                    type: array
+                    items:
+                      type: string
+                    description: An array of strings, any of which MAY be a URI, indicating the available scopes for this resource set. URIs MUST resolve to scope descriptions as defined in Section 2.1. Published scope descriptions MAY reside anywhere on the web; a resource server is not required to self-host scope descriptions and may wish to point to standardized scope descriptions residing elsewhere. It is the resource server's responsibility to ensure that scope description documents are accessible to authorization servers through GET calls to support any user interface requirements. The resource server and authorization server are presumed to have separately negotiated any required interpretation of scope handling not conveyed through scope descriptions.
+                  scopeExpression:
```
Comment (yuriyz) [inline docs/oxAuth_swagger.yaml]: `scope_expression`, https://github.com/GluuFederation/oxAuth/blob/e5ae58e5c6e605f32288549c514d4542454cbb2b/Model/src/main/java/org/gluu/oxauth/model/uma/UmaResource.java#L141

File: docs/oxAuth_swagger.yaml [hunk 15]
```diff
@@ -0,0 +1,3426 @@
+# raw swagger spec link:
+# https://raw.githubusercontent.com/GluuFederation/oxAuth/master/docs/oxAuth_swagger.yaml
+# https://gluu.org/swagger-ui/?url=https://raw.githubusercontent.com/GluuFederation/oxAuth-server/version_4.0/oxAuth/master/docs/oxAuth_swagger.yaml#/developers/setup-client
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Yes makes sense, have modified the link

File: docs/oxAuth_swagger.yaml [hunk 16]
```diff
@@ -0,0 +1,3426 @@
...
+openapi: 3.0.1
+info:
+  title: oxAuth
+  description: oxAuth - Authorization services
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Modified the description

File: docs/oxAuth_swagger.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Added jti property in IntrospectionResponse. 
nbf property not added

File: docs/oxAuth_swagger.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Added jti property in IntrospectionResponse.
nbf property not added

File: docs/oxAuth_swagger.yaml [hunk 17]
```diff
@@ -0,0 +1,3426 @@
...
+      operationId: uma2-configuration    
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                description: UmaMetadata
+                required:
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: - Have rearranged all the properties in order first from OAuth2Discovery followed by properties from UmaMetadata.
- Only code_challenge_methods_supported from OAuth2Discovery was missing and have now added the same.
- Following properties are commented in OAuth2Discovery and hence were not included. 
-- revocation_endpoint,
-- revocation_endpoint_auth_methods_supported, 
-- revocation_endpoint_auth_signing_alg_values_supported,
-- introspection_endpoint_auth_methods_supported, 
-- introspection_endpoint_auth_signing_alg_values_supported
- Please confirm if any other changes required?

File: docs/oxAuth_swagger.yaml [hunk 18]
```diff
@@ -0,0 +1,3426 @@
...
+              schema:
+                title: UmaPermissionList
+                type: array
+                description: List of Uma Permission.
+                items:
+                  type: object
+                  required:
+                  - resource_id
+                  - resource_scopes
+                  properties:
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Added 'exp' property

File: docs/oxAuth_swagger.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Any more changes require?

File: docs/oxAuth_swagger.yaml (hunk 14 above)
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: rectified

File: docs/oxAuth_swagger.yaml [hunk 19]
```diff
@@ -0,0 +1,3426 @@
...
+          required: false
+          description: CIBA authentication request Id.
+          schema:
+            type : string
+      responses:
+        200:
+          description: OK
+          content: {}
+        302:
+          $ref: '#/components/responses/Found'
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Yuriy, bit of confusion here , these ErrorResponseType are for different endpoints and hence how can we use oneOf?
Rather if we have to specify the different error_code can we use enum as follows?
image

File: docs/oxAuth_swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Will implement the ErrorResponseType once the query on previous point is resolved

File: docs/oxAuth_swagger.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Will list the ErrorResponseType once the query on previous point is resolved

File: docs/oxAuth_swagger.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: I agree, lets use enum. `oneOf` produce too much redundancy.

File: docs/oxAuth_swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: Good

File: docs/oxAuth_swagger.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: Agreed

File: docs/oxAuth_swagger.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: Agreed

File: docs/oxAuth_swagger.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: That's it, all properties listed in these 2 beans.

File: docs/oxAuth_swagger.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: no, this part looks good

File: docs/oxAuth_swagger.yaml [hunk 20]
```diff
@@ -0,0 +1,3426 @@
...
+                    type: string
+                    description: Location of the Client Configuration Endpoint where the Registration Access Token can be used to perform subsequent operations upon the resulting Client registration. 
+                  client_id_issued_at:
+                    type: integer
+                    description: Time at which the Client Identifier was issued. 
+                  client_secret_expires_at:
+                    type: integer
+                    description: Time at which the client_secret will expire or 0 if it will not expire.                
+        400:
+          $ref: '#/components/responses/InvalidRequest'
```
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: Done changes, please confirm if this is fine
image

[the remark this answers is not in the document]
Comment (yuriyz) [reply docs/oxAuth_swagger.yaml]: Looks good

File: docs/oxAuth_swagger.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml]: endpoints modified for review :
- /authorize (get & post) -> 302 response error code
-  /bc-authorize -> 401 response error code
- /bc-deviceRegistration -> 400 response error code
- /clientinfo ->(get & post)  401 response code
 **Rest of the oxAuth service will be changed for other ErrorResponseType based on confirmation**

File: docs/oxAuth_swagger.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:227]: If no further changes required then this review can be resolved

File: docs/oxAuth_swagger.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:365]: If no further changes required then this review can be resolved

File: docs/oxAuth_swagger.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:749]: If no further changes required then this review can be resolved

File: docs/oxAuth_swagger.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:876]: If no further changes required then this review can be resolved

File: docs/oxAuth_swagger.yaml (hunk 17 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:2396]: If no further changes required then this review can be resolved

File: docs/oxAuth_swagger.yaml (hunk 18 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:2591]: If no further changes required then this review can be resolved

File: docs/oxAuth_swagger.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pujavs) [reply docs/oxAuth_swagger.yaml:2753]: This can be resolved

**Comment by yuriyz** [comment]
@pujavs I see few commits but no review request. Is it finished or you still working on it?

**Comment by yuriyz** [comment]
No reply from @pujavs , anyway merging it, since new properties has to be added while we developing new features.

**Pull request merged**